File: 05/part2.py
from string import ascii_lowercase

from part1 import react

if __name__ == '__main__':
    with open('input.txt') as f:
        polymer = f.read().strip()
    results = (
        len(react(polymer.replace(letter, '').replace(letter.upper(), '')))
        for letter in ascii_lowercase
    )
    print(min(results))

File: 2018/05/part2.exs
Code.require_file("part1.exs")

defmodule AoC05_2 do
  def improve(polymer) do
    Stream.map(?a..?z, fn c ->
      String.replace(polymer, ~r/#{<<c>>}/i, "", global: true)
      |> AoC05_1.react
      |> String.length
    end)
    |> Enum.min
  end
end

File.stream!("input.txt")
|> Enum.at(0)
|> String.trim
|> AoC05_2.improve
|> IO.puts
